File: SDKs/googleapis.com/securitycenter/v1/python/src/sdk/models/shared/kubernetes.py
import dataclasses
from typing import Optional
from dataclasses_json import dataclass_json
from sdk import utils
from ..shared import accessreview as shared_accessreview
from ..shared import googlecloudsecuritycenterv1binding as shared_googlecloudsecuritycenterv1binding
from ..shared import nodepool as shared_nodepool
from ..shared import node as shared_node
from ..shared import pod as shared_pod
from ..shared import role as shared_role


@dataclass_json
@dataclasses.dataclass
class Kubernetes:
    r"""Kubernetes
    Kubernetes related attributes.
    """
    
    access_reviews: Optional[list[shared_accessreview.AccessReview]] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('accessReviews') }})
    bindings: Optional[list[shared_googlecloudsecuritycenterv1binding.GoogleCloudSecuritycenterV1Binding]] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('bindings') }})
    node_pools: Optional[list[shared_nodepool.NodePool]] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('nodePools') }})
    nodes: Optional[list[shared_node.Node]] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('nodes') }})
    pods: Optional[list[shared_pod.Pod]] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('pods') }})
    roles: Optional[list[shared_role.Role]] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('roles') }})
    

File: SDKs/googleapis.com/securitycenter/v1/java/lib/src/main/java/openapisdk/models/shared/Kubernetes.java
package openapisdk.models.shared;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonInclude.Include;

/**
 * Kubernetes
 * Kubernetes related attributes.
**/
public class Kubernetes {
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("accessReviews")
    public AccessReview[] accessReviews;
    public Kubernetes withAccessReviews(AccessReview[] accessReviews) {
        this.accessReviews = accessReviews;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("bindings")
    public GoogleCloudSecuritycenterV1Binding[] bindings;
    public Kubernetes withBindings(GoogleCloudSecuritycenterV1Binding[] bindings) {
        this.bindings = bindings;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("nodePools")
    public NodePool[] nodePools;
    public Kubernetes withNodePools(NodePool[] nodePools) {
        this.nodePools = nodePools;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("nodes")
    public Node[] nodes;
    public Kubernetes withNodes(Node[] nodes) {
        this.nodes = nodes;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("pods")
    public Pod[] pods;
    public Kubernetes withPods(Pod[] pods) {
        this.pods = pods;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("roles")
    public Role[] roles;
    public Kubernetes withRoles(Role[] roles) {
        this.roles = roles;
        return this;
    }
}